### PR TITLE
doc: release notes 3.6: STM32 changes and shields additions

### PR DESCRIPTION
**Pull request by erwango**

### CODE REVIEW

File: doc/releases/release-notes-3.6.rst [hunk 1]
```diff
@@ -244,36 +284,51 @@ Drivers and Sensors
...
 
 * Display
 
+  * Introduce frame buffer config to STM32 LTDC driver.
+
 * DMA
 
+  * STM32WBA Devices now support GPDMA
+
+* EEPROM
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.6.rst:312]: This needs not be added.

File: doc/releases/release-notes-3.6.rst [hunk 2]
```diff
@@ -339,20 +400,39 @@ Drivers and Sensors
...
   * Renamed ``CONFIG_UART_RA`` to :kconfig:option:`CONFIG_UART_RENESAS_RA`
+  * On STM32 devices, it is now possible to enable FIFO by setting a ``fifo-enable``
+    property in targeted serial node, with the following benefits:
+    In TX, FIFO allows to work in burst mode, easing scheduling of loaded applications.
+    It also allows more reliable communication with UART devices sensitive to variation of inter-frames delays.
+    In RX, FIFO reduces overrun occurences.
 
 * SPI
 
+  * On STM32H7 devices, `fifo-enable` property allows using SPI block FIFO. This
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.6.rst:422]: Needs double quotes.

File: doc/releases/release-notes-3.6.rst [hunk 3]
```diff
@@ -339,20 +400,39 @@ Drivers and Sensors
...
+    featue is still experimental and requires maturation.
+  * On STM32 devices impacted by BSY bit erratum, a workaround is implemented.
+
 * Timer
 
 * USB
 
+  * On STM2G0 devices, property``crs-usb-sof`` in ``clk_hsi48`` node enables support
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.6.rst:430]: Missing space after "property".

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Very comprehensive, thanks.

File: doc/releases/release-notes-3.6.rst [hunk 4]
```diff
@@ -72,23 +72,48 @@ Boards & SoC Support
...
+
+  * Added support for ST STM32L4R9I Discovery board: ``stm32l4r9i_disco``
+
+  * Added support for ST STM32U5A9J-DK discovery kit: ``stm32u5a9j_dk``
+
+  * Added support for ST Nucleo WBA55CG board: ``nucleo_wba55cg``
+
+  * Added support for ST STM32WB5MM-DK Discovery board: ``stm32wb5mm_dk``
+
+  * Added support for FANKE FK7B0M1-VBT6 board: ``fk7b0m1_vbt6``
```
Comment (erwango) [inline doc/releases/release-notes-3.6.rst:116]: I need to fix this

File: doc/releases/release-notes-3.6.rst (hunk 1 above)
Comment (erwango) [reply doc/releases/release-notes-3.6.rst:312]: Yeah, this is coming from a rebase, not sure how it happened.

File: doc/releases/release-notes-3.6.rst [hunk 5]
```diff
@@ -207,6 +224,15 @@ Drivers and Sensors
 
 * ADC
 
+  * Power Management for ADC is now supported on STM32 devices.
+  * STM32 ADC driver now supports mixing shared and separate IRQs (for instance on STM32G473
+    which has 5 ADCs, ADC1 and ADC2 share one IRQ while ADC3, ADC4 and ADC5 each have unique IRQs).
+    Enabling all instances in same ammplication is not possible on such devices as of now.
```
Comment (ajarmouni-st) [inline doc/releases/release-notes-3.6.rst:230]: in same ammplication

File: doc/releases/release-notes-3.6.rst [hunk 6]
```diff
@@ -322,6 +365,8 @@ Drivers and Sensors
 
   * Renesas R-Car pinctrl driver now supports Gen4 SoCs
   * Renamed ``CONFIG_PINCTRL_RA`` to :kconfig:option:`CONFIG_PINCTRL_RENESAS_RA`
+  * On STM32 devices, pins assigned to JTAG/SW port can now be put to analog state when
+    :kconfig:option:`CONFIG_PM=y` and :kconfig:option:`CONFIG_DEBUG=n`
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.6.rst:369]: You can't have the `=y` and `=n` parts in the references.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.6.rst (hunk 6 above)
Comment (erwango) [reply doc/releases/release-notes-3.6.rst:369]: TIL

**Review by henrikbrixandersen — DISMISSED** [review]

File: doc/releases/release-notes-3.6.rst [hunk 7]
```diff
@@ -207,6 +218,15 @@ Drivers and Sensors
 
 * ADC
 
+  * Power Management for ADC is now supported on STM32 devices.
+  * STM32 ADC driver now supports mixing shared and separate IRQs (for instance on STM32G473
+    which has 5 ADCs, ADC1 and ADC2 share one IRQ while ADC3, ADC4 and ADC5 each have unique IRQs).
+    Enabling all instances in same ammplication is not possible on such devices as of now.
```
Comment (MaureenHelm) [inline doc/releases/release-notes-3.6.rst:230]:
```suggestion
    Enabling all instances in same application is not possible on such devices as of now.
```

File: doc/releases/release-notes-3.6.rst [hunk 8]
```diff
@@ -244,36 +264,49 @@ Drivers and Sensors
 
   * Renesas R-Car clock control driver now supports Gen4 SoCs
   * Renamed ``CONFIG_CLOCK_CONTROL_RA`` to :kconfig:option:`CONFIG_CLOCK_CONTROL_RENESAS_RA`
+  * On STM32 devices, :dtcompatible:`st,stm32-hse-clock` no allows setting a ``css-enabled``
```
Comment (MaureenHelm) [inline doc/releases/release-notes-3.6.rst:273]:
```suggestion
  * On STM32 devices, :dtcompatible:`st,stm32-hse-clock` now allows setting a ``css-enabled``
 ```

**Review by MaureenHelm — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by erwango** [comment]
@MaureenHelm, @henrikbrixandersen sorry I had to rebase

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]
very nicely done, thanks!

**Review by msmttchr — APPROVED** [review]
Thanks

**Review by MaureenHelm — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
@erwango More merge conflicts, I'm afraid.

**Comment by erwango** [comment]
@MaureenHelm, @kartben, @henrikbrixandersen rebase done.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by ajarmouni-st — CHANGES_REQUESTED** [review]

**Pull request merged**